jobsdb/jobsdb.go

```diff
@@ -226,26 +226,26 @@ type JobsDB interface {
 
 	// GetUnprocessed finds unprocessed jobs. Unprocessed are new
 	// jobs whose state hasn't been marked in the database yet
-	GetUnprocessed(params GetQueryParamsT) JobsResult
+	GetUnprocessed(ctx context.Context, params GetQueryParamsT) (JobsResult, error)
```

Comment:
I believe need at least one integration test where we will actually be using the database for verifying that indeed the `sql` package propagates the `context.DeadlineExceeded` error as expected:

e.g. 
1. Start jobsDB
2. Add some jobs
3. Explicitly lock the jobs table and keep the lock until the end
4. Call GetUnprocessed with a context expiring in 1 second
5. Verify that the error is a `context.DeadlineExceeded` one